File: problem20/problem.py
def sumAllNumberDigits(number, acc=0):
    if number > 0:
        return acc + sumAllNumberDigits(number / 10, number % 10)
    else:
        return acc


def factorial(number, acc=1):
    if (number == 1):
        return acc
    else:
        return factorial(number - 1, acc * number)

File: javascript/problem8/problem.js
export const getSubString = (inputDigit, fromIndex, ofLength) => {
    return inputDigit.substr(fromIndex, ofLength)
}

export const multiplyDigitsInNumber = (numberString) => {
    return numberString
    .split('')
    .map(number => parseInt(number))
    .reduce((acc, current) => acc * current, 1)
}

export const getMaxMultiplicationValueOfSubstring = (inputDigit, numDigits) => {
    let fromIndex = 0
    let maxValue = 0
    const inputLenght = inputDigit.length

    while (fromIndex < inputLenght - numDigits + 1) {
        let subDigit = getSubString (inputDigit, fromIndex, numDigits)
        const multiplication = multiplyDigitsInNumber(subDigit)
        maxValue = Math.max(maxValue, multiplication)
        fromIndex = fromIndex + 1
    }
    return maxValue
}